backend/apps/github/common.py

```diff
@@ -13,9 +17,68 @@
 from apps.github.models.user import User
 from apps.github.utils import check_owasp_site_repository
 
+INVALID_ISSUE_LINK_FORMAT = "Invalid GitHub issue link format."
+FETCH_ISSUE_ERROR = "Failed to fetch issue from GitHub: {error}"
+MIN_PARTS_LENGTH = 4
+
+
 logger = logging.getLogger(__name__)
 
 
+def sync_issue(issue_link):
+    """Sync GitHub issue data."""
+    try:
+        return Issue.objects.get(url=issue_link)
+    except Issue.DoesNotExist:
+        pass
+
+    parsed_url = urlparse(issue_link)
+    path_parts = parsed_url.path.strip("/").split("/")
+    if len(path_parts) < MIN_PARTS_LENGTH or path_parts[2] != "issues":
+        raise ValidationError(INVALID_ISSUE_LINK_FORMAT)
```

Comment:
The validation should happen upon saving the model instance. This function should accept a clean link w/o worrying about validation errors.